apps/clientside-bot/test/functions.test.ts

```diff
@@ -0,0 +1,1070 @@
+					communityIds: guildConfigs.trustedCommunities,
+					// to ensure that the playername is identical
+					playernames: [faker.internet.userName()],
+				},
+			],
+			2
+		)
+
+		await database.getRepository(FAGCBan).insert(oldReport)
+
+		const results = await handleReport({
+			report: newReport,
+			database,
+			allGuildConfigs: [guildConfigs],
+		})
+
+		const foundInDatabase = await database.manager.find(FAGCBan)
+
+		// should be an array of guild IDs rather than false
+		expect(results).not.toBe(false)
+		expect(results).toBeInstanceOf(Array)
+		// the array should be blank, as the player is banned already
+		expect(results).toEqual([])
+		// there should be two bans in the database, which are equal to the simplified reports
+		expect(foundInDatabase.length).toBe(2)
+		expect(foundInDatabase).toEqual([
+			reportIntoFAGCBan(oldReport),
+			reportIntoFAGCBan(newReport),
+		])
+	})
+	it("Should not ban a player if they are whitelisted", async () => {
+		const guildConfigs = createGuildConfig({
+			categoryIds,
+			communityIds,
+		})
+
+		const report = createFAGCReport({
+			categoryIds: guildConfigs.categoryFilters,
+			communityIds: guildConfigs.trustedCommunities,
+		})
+
+		await database.getRepository(Whitelist).insert({
+			playername: report.playername,
+			adminId: "123",
+		})
+
+		const results = await handleReport({
+			report,
+			database,
+			allGuildConfigs: [guildConfigs],
+		})
+
+		const foundInDatabase = await database.manager.find(FAGCBan)
+
+		// should be an array of guild IDs rather than false
+		expect(results).not.toBe(false)
+		expect(results).toBeInstanceOf(Array)
+		// the array should be blank, as the player is ignored
+		expect(results).toEqual([])
+		// there should be a ban in the database, which is equal to the simplified report
+		expect(foundInDatabase.length).toBe(1)
+		expect(foundInDatabase[0]).toEqual(reportIntoFAGCBan(report))
+	})
+	it("Should not ban a player if they are private banned", async () => {
+		const guildConfigs = createGuildConfig({
+			categoryIds,
+			communityIds,
+		})
+
+		const report = createFAGCReport({
+			categoryIds: guildConfigs.categoryFilters,
+			communityIds: guildConfigs.trustedCommunities,
+		})
+
+		await database.getRepository(PrivateBan).insert({
+			playername: report.playername,
+			adminId: "123",
+		})
+
+		const results = await handleReport({
+			report,
+			database,
+			allGuildConfigs: [guildConfigs],
+		})
+
+		const foundInDatabase = await database.manager.find(FAGCBan)
+
+		// should be an array of guild IDs rather than false
+		expect(results).not.toBe(false)
+		expect(results).toBeInstanceOf(Array)
+		// the array should be blank, as the player is ignored
+		expect(results).toEqual([])
+		// there should be a ban in the database, which is equal to the simplified report
+		expect(foundInDatabase.length).toBe(1)
+		expect(foundInDatabase[0]).toEqual(reportIntoFAGCBan(report))
+	})
+})
+describe("handleRevocation", () => {
```

Comment:
Keep consistent whitespace.
```suggestion
})

describe("handleRevocation", () => {
```